src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotDefineFinalizersForTypesDerivedFromMemoryManagerTests.cs

```diff
@@ -0,0 +1,229 @@
+                    Inherits MemoryManager(Of T)
+                End Class
+            End Namespace
+            ");
+        }
+
+        [Fact]
+        public async Task ClassDerivedFromMemoryManagerWithFinilizerWarns()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+            using System;
+            using System.Buffers;
+
+             namespace TestNamespace
+             {
+                class TestClass<T> : MemoryManager<T>
+                {
+                    public override Span<T> GetSpan()
+                    {
+                        throw new NotImplementedException();
+                    }
+
+                    public override MemoryHandle Pin(int elementIndex = 0)
+                    {
+                        throw new NotImplementedException();
+                    }
+
+                    public override void Unpin() { }
+
+                    ~TestClass() { }
+
+                    protected override void Dispose(bool disposing) { }
+                }
+            }",
+                GetCSharpWarningResultAt(21, 22));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+            Imports System
+            Imports System.Buffers
+
+            Namespace TestNamespace
+                MustInherit Class TestClass(Of T)
+                    Inherits MemoryManager(Of T)
+
+                    Public Overrides Function Pin(ByVal Optional elementIndex As Integer = 0) As MemoryHandle
+                        Throw New NotImplementedException()
+                    End Function
+
+                    Public Overrides Sub Unpin()
+                    End Sub
+
+                    Protected Overrides Sub Finalize()
+                    End Sub
+
+                    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
+                    End Sub
+                End Class
+            End Namespace",
+                GetVisualBasicWarningResultAt(16, 45));
+        }
+
+        [Fact]
+        public async Task ClassIndirectlyDerivedFromMemoryManagerWithFinilizerWarns()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+            using System;
+            using System.Buffers;
+
+             namespace TestNamespace
+             {
+                class Deeper<T> : Middle<T>
+                {
+                    ~Deeper() { }
+                }
+
+                class Middle<T> : MemoryManager<T>
+                {
+                    public override Span<T> GetSpan()
+                    {
+                        throw new NotImplementedException();
+                    }
+
+                    public override MemoryHandle Pin(int elementIndex = 0)
+                    {
+                        throw new NotImplementedException();
+                    }
+
+                    public override void Unpin() { }
+
+                    protected override void Dispose(bool disposing) { }
+                }
+            }",
+                GetCSharpWarningResultAt(9, 22));
```

Comment:
💡 Prefer markup syntax for tests so it's easier to see where the diagnostic is reported in code. If you want to test other specifics of the diagnostics (message, arguments, severity, and/or additional spans), make one test (or a limited few) covering those aspects.

Reply:
Not sure what ` markup syntax` please share some explanation or sample 🙏